src/FSharpVSPowerTools.Core/Utils.fs

```diff
@@ -545,14 +700,18 @@ module Dict =
 [<RequireQualifiedAccess>]
 [<CompilationRepresentation(CompilationRepresentationFlags.ModuleSuffix)>]
 module String =
+
+    let inline toCharArray (str:string) = str.ToCharArray()
+
     let lowerCaseFirstChar (str: string) =
-        match str with
-        | null -> null
-        | str ->
-            match str.ToCharArray() |> Array.toList with
-            | [] -> str
-            | h :: t when Char.IsUpper h -> String (Char.ToLower h :: t |> List.toArray)
-            | _ -> str
+        if isNull str then null else 
```

Comment:
This pattern (`if isNull ... then null else ...`) repeats quite often now. What about:

```fsharp
[<RequireQualifiedAccess>]
module Null =
    let inline fill defaultValue x = if isNull x then null else defaultValue
    let inline fillWith (genDefaultValue: unit -> 'a) x = if isNull x then null else genDefaultValue()
```
Then you can rewrite the code like this:
```fsharp
str |> Null.fill (fun _ -> ...)
```